src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -233,9 +233,12 @@ internal void RemoveProject(AbstractProject project)
 
             UpdateProjectBinPath(project, project.TryGetBinOutputPath(), null);
 
-            if (project.PushingChangesToWorkspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
             {
-                NotifyWorkspaceHosts(host => host.OnProjectRemoved(project.Id));
+                foreach (var hostState in _workspaceHosts)
```

Comment:
Then hostState.RemoveProject will no-op because there aren't any projects pushed. The list is empty, Remove will do nothing.